sys/include/net/gnrc/ipv6/nib/ft.h

```diff
@@ -68,14 +68,16 @@ int gnrc_ipv6_nib_ft_get(const ipv6_addr_t *dst, gnrc_pktsnip_t *pkt,
  *                      default route.
  * @param[in] next_hop  The next hop to @p dst/@p dst_len. May be NULL, if
  *                      @p dst/@p dst_len is no the default route.
+ * @param[in] ltime     Lifetime in seconds of this route entry.
  * @param[in] iface     The interface to @p next_hop. May not be 0.
  *
  * @return  0, on success.
  * @return  -EINVAL, if a parameter was of invalid value.
  * @return  -ENOMEM, if there was no space left in forwarding table.
  */
 int gnrc_ipv6_nib_ft_add(const ipv6_addr_t *dst, unsigned dst_len,
-                         const ipv6_addr_t *next_hop, unsigned iface);
+                         const ipv6_addr_t *next_hop, uint16_t ltime,
```

Comment:
This was kept out of the intentionally. The routing protocol should timeout the routes externally (e.g. by storing the context for the timer in the protocol's routing table)

Reply:
This is weird. Every FIB entry should have a lifetime. RPL is not aware of link failures way down the tree as it does not maintain a children list, only a parent list. So RPL does not know when to prune an entry in that case. FIB maintenance should be in the FIB, not in a routing protocol.

Reply:
This isn't about FIB maintenance, but about route maintenance IMHO. The lifetime of the route (and its scale) is intrinsic to the routing protocol (if you want a proof for that, see various discussions in our GitHub issue history about which unit should be used with route lifetimes).

Reply:
Iff we do it like this I have some questions: Why is the route lifetime in seconds (isn't milliseconds a more usual unit for that)? Why did you choose `uint16_t` as the type for that?

Reply:
>Why is the route lifetime in seconds (isn't milliseconds a more usual unit for that)? Why did you choose uint16_t as the type for that?

We had the discussion for `gnrc_fib` that milliseconds granularity is overkill and the port to deciseconds was never done.. I guess having a second granularity is okay for our constrained IoT scenario and routing protocols usually are not that fast in noticing routing state changes.

Reply:
Oh, and I chose `uint16_t` because that's what the default router struct is also using.